app/controllers/smart_proxies_controller.rb

```diff
@@ -63,8 +66,12 @@ def destroy
 
   private
 
-  def requested_data(data_name)
-    data = @smart_proxy.public_send(data_name)
+  def find_status
+    @proxy_status = @smart_proxy.proxy_status
+  end
+
+  def requested_data(object, method)
+    data = object.public_send(method)
     render :json => {:success => true, :message => data}
```

Comment:
just a thought - this is basically a wrapper. So instead of passing an object and method, how about something along the lines of:
```
def requested_data()
  render :json => {:success => true, :message => yield}
rescue Foreman::Exception => exception
  render :json => {:success => false, :message => exception.message} and return
end
```